ingestion-beam/src/main/java/com/mozilla/telemetry/decoder/ParseProxy.java

```diff
@@ -48,14 +48,16 @@ public PubsubMessage apply(PubsubMessage message) {
       final String xpp = attributes.get(Attribute.X_PIPELINE_PROXY);
       String xff = attributes.get(Attribute.X_FORWARDED_FOR);
 
-      // If the configured proxyIpAddress is present, then there's a GCP load balancer IP
+      // If any of the configured proxyIpAddresses is present, then there's a GCP load balancer IP
       // we will need to remove; see
       // https://bugzilla.mozilla.org/show_bug.cgi?id=1729069#c8
-      if (xff != null && xff.contains(proxyIpAddress)) {
-        xff = Arrays.stream(xff.split("\\s*,\\s*")) //
-            .filter(ip -> !proxyIpAddress.equals(ip)) //
-            .collect(Collectors.joining(","));
-        countProxyIpAddress.inc();
+      for (String proxyIpAddress : proxyIpAddressCsv.split(",")) {
```

Comment:
I think we should parse this parameter into a `Set<String>` within the constructor. Then we can check `ip -> !proxyIpAddresses.contains(ip)` and avoid having the outer for loop.

We don't do that sort of thing in existing code because we only recently transitioned from "classic templates" to Flex Templates. With classic templates, this would have been a runtime parameter that would not have been defined when the constructor was called. But now all pipeline options are fully available when the code is compiled.

Reply:
How would `countProxyIpAddress.inc()` work in this case? Is it not counting the number of proxy ips that get matched? (Previously there would only have been one)

Reply:
Good question. We'd only ever expect one proxy IP to be present in `xff`. I think it's fine to just increment the counter once based on `proxyIpAddresses.contains(ip)` being true.

Reply:
I have a version that removes the outer loop. Something like:
```
private final Set<String> proxyIps;
...
if (xff != null) { // && at least one ip in the xff appears in the proxyIps {
        xff = Arrays.stream(xff.split("\\s*,\\s*")) //
            .filter(StringUtils::isNotBlank) //
            .filter(ip -> !proxyIps.contains(ip)) //
            .collect(Collectors.joining(","));
        countProxyIpAddress.inc();
 }
```
The extra clause would be something like "if the set of xff ips intersects the set of `proxyIps`" since each `ip` is only in scope in the closures unless I'm missing something... Is that what you had in mind?